### ISSUE TITLE
Webelements don't contain text properties when it's a Web Hybrid app with ionic

### ISSUE THREAD

**Issue by CarlosNano** [appium/appium#16436]
## The problem

We are trying to test a Web Hybrid app made with ionic 

When using Appium inspector ( the result in my tests with AppiumDriver is the same ) and choosing the web context, it's not possible to get any text from labels such the h2 in the example or the text in the input.

![image](https://user-images.githubusercontent.com/72921935/153387069-2f5aecdc-af25-4c39-9de4-b6b1fb45566a.png)
![image](https://user-images.githubusercontent.com/72921935/153387411-921171d2-f1ab-4fd2-beac-648c0b9a4deb.png)

I also checked in the Chrome dev tools that the text is there.

![image](https://user-images.githubusercontent.com/72921935/153391545-f189e93b-d19b-4e57-be9c-d596cd914cfc.png)

If I change to the Native context, most of the labels are easy to get. But I should not work with Native context because it's a web Hybrid app and also the structure is different in the views/`<div>`

Capabilities in the Appium Driver:

            caps.setCapability("platformName", "Android");
            caps.setCapability("platformVersion", "11");
            caps.setCapability("deviceName", "emulator-5554");
            caps.setCapability("automationName", "UiAutomator2");
            caps.setCapability("udid", "emulator-5554");
            caps.setCapability("app",androidAppPath);
            caps.setCapability("fullContextList", true);
            driver = new AndroidDriver(appiumUrl, caps);

## Environment

* Appium version: 1.22.0
* Desktop OS/version used to run Appium: macOS 10.15.7
* Node.js version: v14.17.5
* Npm or Yarn package manager: npm 
* Mobile platform/version under test: Android 11
* Real device or emulator/simulator: Simulator
* Appium CLI or Appium.app|exe: Appium App

I also checked the version of ChromeDriver and the version of my Chrome are similar (except for the last three of numbers)

Chome version: 83.0.4103.XXX

## Link to Appium logs

https://gist.github.com/CarlosNano/6bc077ce625e9c2af386c66185ebb9f2

Am I missing something here? Any help would be very appreciated.
Thanks

**Comment by mykola-mokhnach** [comment]
Is the destination element present while retrieving the page source from chromedriver?

**Comment by CarlosNano** [comment]
Thanks for the quick answer, how can I get the page source from the chromedriver?

I know I'm able to do:
```
            System.out.println("All:" + driver.getPageSource());
```

But that `driver` it's a `AndroidDriver`

The `<input>` is in the page source:
```
<input placeholder="Memorable name" class="editable sc-app-device-card-header" id="com.myCompany.myAppName:id/device-card-header-input">
```

**Comment by mykola-mokhnach** [comment]
the page source is returned per context, e.g. if you call it being in chrome context then the page source of the web view is returned. So again, if the text present present in that page source or not?

**Comment by mykola-mokhnach** [comment]
Then this means chromedriver itself does not get it. In such case only Google is capable of fixing it.

**Comment by CarlosNano** [comment]
Ok I think I found how to get it for the input.

In the chrome context:
```
            private By inputBy = By.xpath(".//*[@id=\"myId\"]");

            WebElement input = view.findElement(this.inputBy);

            String value = input.getAttribute("value");
```
But not sure if there is a bug why `getText` in the WebElement doesn't work.

Thanks for the help @mykola-mokhnach

**Comment by CarlosNano** [comment]
Yeah, I thought that too, what about the labels? is normal a `webElement` pointing to a label doesn't provide the text in the `getText()` function.

Thanks :)

**Comment by mykola-mokhnach** [comment]
Web and native context are different. The best way to see what's available is to check the page source

**Comment by mykola-mokhnach** [comment]
Closed as not an issue
